components/learn/modules/ROOT/pages/public-staging.adoc

```diff
@@ -1,39 +1,183 @@
+const HDWalletProvider = require('@truffle/hdwallet-provider');
+
+module.exports = {
+  networks: {
+    development: { ... },
+    rinkeby: {
+      provider: () => new HDWalletProvider(mnemonic, `https://rinkeby.infura.io/v3/${projectId}`),
+      networkId: 4,
+      gasPrice: 10e9
+    }
+  },
+};
+----
+
+NOTE: The `HDWalletProvider` can be configured to generate more than a single account from the mnemonic. Check out its https://github.com/trufflesuite/truffle/tree/master/packages/hdwallet-provider[documentation] for more info.
+
+Note in the first line that we are loading the project id and mnemonic from a `secrets.json` file, which should look like the following, but using your own values. Make sure to `.gitignore` it!
+
+[source,json]
+----
+{ 
+  "mnemonic": "mistake pelican mention moment ...", 
+  "projectId": "305c13705054a8d918ad77549e402c72"
+}
+----
+
+NOTE: Instead of a `secrets.json` file, you can use whatever secret-management solution you like for your project. A popular and simple option is to use https://github.com/motdotla/dotenv[`dotenv`] for injecting secrets as environment variables.
+
+We can now test out that this configuration is working by listing the accounts we have available for the Rinkeby network. Remember that yours will be different, as they depend on the mnemonic you used.
+
+[source,console]
+----
+$ npx oz accounts
+? Pick a network: rinkeby
+Accounts for rinkeby:
+Default: 0xf0A9eD2663311CE436347Bb6F240181FF103CA16
+All:
+- 0: 0xf0A9eD2663311CE436347Bb6F240181FF103CA16
+- 1: 0x3B9861c7D3e7BBd41602d9FfaCEF10BC04867Bc0
+- 2: 0x8C7623AC7Fe2E635Fa256791C25dA2c8851c5F08
+- 3: 0xd86f3FeeFd93bd19acaFd212D8630DEDeb56C6bd
+...
+----
+
+We can also test the connection to the Infura node, by querying our account balance.
+
+[source,console]
+----
+$ npx oz balance
+? Enter an address to query its balance: 0xf0A9eD2663311CE436347Bb6F240181FF103CA16
+? Pick a network: rinkeby
+Balance: 0 ETH
+----
+
+Empty! This points to our next task: getting testnet funds for sending transactions from our account.
+
+=== Funding the Testnet Account
+
+Most public testnets have a faucet: a site that will provide you with a small amount of test Ether for free. If you are on Rinkeby, head on to the https://faucet.rinkeby.io/[Rinkeby Authenticated Faucet] to get funds by authenticating with your twitter or facebook account. Alternatively, you can also use https://faucet.metamask.io/[Metamask's faucet] to ask for funds directly to your Metamask accounts.
+
+Now, armed with a funded account, let's deploy our contracts to the testnet.
+
+[[deploy-and-interact]]
+== Working on a Testnet
+
+With a project configured to work on a public testnet, we can now finally xref::deploy-and-interact.adoc#box-contract[deploy our `Box` contract]. The command here is exactly the same as if you were on your xref::deploy-and-interact.adoc#local-blockchain[local development network], though it will take few more seconds to run.
```

Comment:
```suggestion
With a project configured to work on a public testnet, we can now finally xref::deploy-and-interact.adoc#box-contract[deploy our `Box` contract]. The command here is exactly the same as if you were on your xref::deploy-and-interact.adoc#local-blockchain[local development network], though it will take a few seconds to run as new blocks are mined.
```